# HG changeset patch
# User Ederag  <edera@gmx.fr>
# Date 1450490228 18000
#      Fri Dec 18 20:57:08 2015 -0500
# Node ID 249a4659dfa5bd91e22f59ebd6c945bc02df814c
# Parent  9db35d2042be7780998302f5f4ffdca1096af79f
Place gnuplot ticklabel exponents for LaTeX terminals in mathmode (bug #46517).

* __gnuplot_draw_axes__.m: For LaTeX related terminals (i.e. cairolatex,
epslatex, latex, pslatex, pstex, pstricks, texdraw, & tikz) place tick labels
with exponents in mathmode, $10^{%T}$.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -2162,17 +2162,18 @@ function do_tics_1 (ticmode, tics, mtics
   elseif (strcmpi (interpreter, "latex"))
     if (! warned_latex)
       warning ("latex markup not supported for tick marks");
       warned_latex = true;
     endif
   endif
   if (strcmp (scale, "log"))
     num_mtics = 10;
-    if (any (strcmp (gnuplot_term, {"tikz", "pstex", "pslatex", "epslatex"})))
+    if (any (strcmp (gnuplot_term, {"cairolatex", "epslatex", "latex", ...
+        "pslatex", "pstex", "pstricks", "texdraw", "tikz"})))
       fmt = "$10^{%T}$";
     else
       fmt = "10^{%T}";
     endif
     if (sgn < 0)
       fmt = ["-" fmt];
     endif
   else
